### DISCUSSION TITLE
how to use pl to process tfrecords data?

### DISCUSSION THREAD

**Discussion by jaffe-fly** [Lightning-AI/pytorch-lightning#8161] [code help: CV] [answered]
how to use ```LightningDataModule```  to process tfrecords data, anyone can give a Tutorial

**Comment by tshu-w** [comment]
How you use python and pytorch to handle tfrecords data is how you use it in `LightningDataModule`. pytorchlightning is just a wrapper.

**Reply by jaffe-fly** [reply]
i know this . so how to process tfrecords with pytorch efficently

**Reply by tshu-w** [reply]
~~... You should ask Google, are the instructions for this package not clear?~~ https://github.com/vahidk/tfrecord#usage

**Comment by Borda** [comment, marked as answer]
Depends on what data you have inside TFRecord, but you can see usage:
 https://github.com/vahidk/tfrecord#reading-tfexample-records-in-pytorch

```py
def train_dataloader():
    # index_path = None
    # tfrecord_path = "/tmp/data.tfrecord"
    # description = {"image": "byte", "label": "float"}
    dataset = TFRecordDataset(tfrecord_path, index_path, description)
    loader = torch.utils.data.DataLoader(dataset, batch_size=32)
    return loader
```

also check how to work with dataset - https://discuss.pytorch.org/t/read-dataset-from-tfrecord-format/16409/15
